libraries/cyclestreets-view/src/main/java/net/cyclestreets/routing/Route.kt

```diff
@@ -194,9 +194,9 @@ object Route {
     }
 
     private fun restoreWaypoints() {
-        val stash = prefs().getString(waypointsInProgressPref, null)
-        stash?.let {
-            waypoints_ = Waypoints(RouteDatabase.deserializeWaypoints(it))
+        val stash = prefs().getString(waypointsInProgressPref, "")
+        if (stash.isNotEmpty()) {
```

Comment:
If stash *is* null - e.g. because someone's run the previous version, where it would default to null - won't we now have an NPE?

I suggest checking both in this `if` test for safety.

Reply:
I don't believe it can be null - it might be the empty string, or it might not be set at all (in which case we now default to the empty string) - but I buggered it up last time, so yes, you're right.